crates/oxc_allocator/src/vec2/raw_vec.rs

```diff
@@ -113,7 +122,7 @@ impl<'a, T> RawVec<'a, T> {
                 }
             };
 
-            RawVec { ptr, a, cap, len: 0 }
+            RawVec { ptr, a, cap: usize_to_u32(cap), len: 0 }
```

Comment:
Similar problem to `Vec::reserve`. You call `Vec::with_capacity_in` with size `n` and you get back a `Vec` with a smaller capacity than you asked for. It should panic instead.

Probably `alloc_guard` is the place for this check.

Note: On 32 bit systems (e.g. WASM), max capacity is `i32::MAX` bytes (half of `u32::MAX`) because Rust requires that no allocation can exceed `isize::MAX` bytes (and on 32-bit, `isize::MAX == i32::MAX`).

Reply:
If I understand correctly, we should change `alloc_guard` to check `alloc_size > ::core::u32::MAX` rather than `alloc_size > ::core::isize::MAX`, right?

https://github.com/oxc-project/oxc/blob/27a65f7a8e9de914279e43d2723b19fe0cf7ede3/crates/oxc_allocator/src/vec2/raw_vec.rs#L752-L759